src/include/transaction/transaction_defs.h

```diff
@@ -18,5 +20,6 @@ class TransactionContext;
 using TransactionQueue = std::forward_list<transaction::TransactionContext *>;
 using callback_fn = void (*)(void *);
 
+using constraint_fn = std::function<bool()>;  // TODO(Yashwanth): look into what parameters this should take.
```

Comment:
I think it should probably take either an entire `TransactionContext` or take individual `RedoRecords`.  Personally, I would go for `RedoRecords` as that should reduce boilerplate code inside the functions and show the logic explicitly in the `CheckConstraint` function.